src/core/mac/mac.cpp

```diff
@@ -822,25 +822,6 @@ TxFrame *Mac::PrepareBeacon(void)
 
     beaconPayload = reinterpret_cast<BeaconPayload *>(beacon->GetPayload());
 
-    if (Get<KeyManager>().GetSecurityPolicy().mBeaconsEnabled)
```

Comment:
`Reserved; defaults to '0'. A Thread Device MUST ignore this bit.` in 1.2 spec, if `ignore`, when sending beacon response, here should have the beacon payload including networkname, extpanid.

Here may directly removes the outer `if` condition to ignore this bit?

Reply:
Notice that section _IEEE 802.15.4 Beacon Frame Format_ has been updated and now:

> The Beacon payload SHOULD be elided though it MAY have an application-specific value.

Reply:
@librasungirl @jwhui @abtink @simonlingoogle this change (removal of Thread Beacons payload) is quite important and affects to many places in `openthread` and `wpantund` (maybe other places in docs, `pyspinel`, etc. as well). Could you confirm we are okay with removing the functionality before I proceed further with this PR?

Reply:
I am OK to remove Thread beacon payload support from OpenThread.

Reply:
confirmed no concern from internal team. also sgtm